app/models/theme_field.rb

```diff
@@ -260,8 +260,8 @@ def process_html(html)
       javascript_cache.source_map = js_compiler.source_map
       javascript_cache.save!
 
-      doc.add_child(<<~HTML.html_safe) if javascript_cache.content.present?
-        <script defer src='#{javascript_cache.url}' data-theme-id='#{theme_id}' nonce="#{CSP_NONCE_PLACEHOLDER}"></script>
+      doc.add_child(<<~HTML.html_safe)
+        <link rel="modulepreload" href="#{javascript_cache.url}" data-theme-id="#{theme_id}" nonce="#{CSP_NONCE_PLACEHOLDER}">
```

Comment:
```suggestion
        <link rel="modulepreload" href="#{javascript_cache.url}" data-theme-id="#{theme_id}" nonce="#{CSP_NONCE_PLACEHOLDER}" />
```

it should be closed after all (looking at the other examples)